stdlib/messaging/kafka/src/main/ballerina/src/kafka/producer.bal

```diff
@@ -0,0 +1,149 @@
+public type Producer client object {
+
+    public ProducerConfig? producerConfig = ();
+
+    public function __init(ProducerConfig config) {
+        self.producerConfig = config;
+        var result = self.init(config);
+        if (result is error) {
+            panic result;
+        }
+    }
+
+    # Initialize the producer endpoint. Panics if the initialization fails.
+    #
+    # + config - Configurations related to the endpoint.
+    # + return - `kafka:ProducerError` if fails to initiate the `kafka:Producer`, nil otherwise.
+    function init(ProducerConfig config) returns error? = external;
+
+    public string connectorId = system:uuid();
+
+    # Closes producer connection to the external Kafka broker.
+    #
+    # + return - `kafka:ProducerError` if closing the producer failed, nil otherwise.
+    public remote function close() returns ProducerError? = external;
+
+    # Commits consumer action which commits consumer consumed offsets to offset topic.
+    #
+    # + consumer - Consumer which needs offsets to be committed.
+    # + return - `kafka:ProducerError` if committing the consumer failed, nil otherwise.
+    public remote function commitConsumer(Consumer consumer) returns ProducerError? = external;
+
+    # CommitConsumerOffsets action which commits consumer offsets in given transaction.
+    #
+    # + offsets - Consumer offsets to commit for given transaction.
+    # + groupID - Consumer group id.
+    # + return - `kafka:ProducerError` if committing consumer offsets failed, nil otherwise.
+    public remote function commitConsumerOffsets(PartitionOffset[] offsets, string groupID) returns ProducerError? = external;
+
+    # Flush action which flush batch of records.
+    #
+    # + return - `kafka:ProducerError` if records couldn't be flushed, nil otherwise.
+    public remote function flushRecords() returns ProducerError? = external;
+
+    # GetTopicPartitions action which returns given topic partition information.
+    #
+    # + topic - Topic which the partition information is given.
+    # + return - `kafka:TopicPartition` array for the given topic, returns `kafka:ProducerError` if operation fails.
+    public remote function getTopicPartitions(string topic) returns TopicPartition[]|ProducerError = external;
+
+    # Simple Send action which produce records to Kafka server.
+    #
+    # + value - Record contents.
+    # + topic - Topic to which the record will be appended to.
+    # + key - Key that will be included in the record.
+    # + partition - Partition to which the record should be sent.
+    # + timestamp - Timestamp of the record, in milliseconds since epoch.
+    # + return - Returns `kafka:ProducerError` if send action fails to send data, nil otherwise.
+    public remote function send(byte[] value, string topic, public byte[]? key = (), public int? partition = (),
+                                public int? timestamp = ()) returns ProducerError? = external;
```

Comment:
If this is a public time duration parameter, shouldn't we update this according to the convention?

Reply:
This is a timestamp value, it isn't a time duration parameter